tests/ignite/engine/test_engine.py

```diff
@@ -935,3 +935,36 @@ def test_is_done_with_max_iters():
 
     state = State(iteration=250, epoch=1, max_epochs=3, epoch_length=100, max_iters=250)
     assert Engine._is_done(state)
+
+
+@pytest.mark.skipif(torch.cuda.device_count() < 1, reason="Skip if no GPU")
+def test_batch_is_released_before_new_one_is_loaded_cuda():
```

Comment:
We have a sort of rule to run non distributed tests on GPU, test should have `on_cuda` in the name (probably something to change as well)
```suggestion
def test_batch_is_released_before_new_one_is_loaded_on_cuda():
```